provider/provider_python_test.go

```diff
@@ -0,0 +1,61 @@
+// Copyright 2016-2023, Pulumi Corporation.  All rights reserved.
+
+//go:build !go && !yaml && !nodejs && !dotnet && !java
+// +build !go,!yaml,!nodejs,!dotnet,!java
+
+package provider
+
+import (
+	"path/filepath"
+	"testing"
+	"time"
+
+	"github.com/pulumi/pulumi/pkg/v3/testing/integration"
+)
+
+func TestRegress3196(t *testing.T) {
+	if testing.Short() {
+		t.Skipf("Skipping test in -short mode because it needs cloud credentials")
+		return
+	}
+	maxDuration(3*time.Minute, t, func(t *testing.T) {
```

Comment:
My assumption is that this would fail before, since the retries would take longer than 3 minutes. Is that right?

Reply:
Yeah like 60 minutes.